### PR TITLE
Explorer Settings Side Panel

### PR DESCRIPTION
**Pull request by P0ppoff**
## Checklist

Definition Of Review

-   [x] Own code review done (add notes for others)
-   [x] Write message in teams channel
    ```
     <Title>

    *️⃣ Impacted projects : Core - DataStudio - Admin - @leav/ui - @leav/utils - ...

    📖 Ticket: https://aristid.atlassian.net/browse/<JIRA_TICKET_IDENTIFIER>

    🧑‍💻 PR: <link to PR/MR>

    ℹ Info: <brief explanation - context - how to test>
    ```

Definition Of Mergeable

-   [x] 2 approves
-   [x] 1 functional review - US has been tested
-   [x] Every comment is handled - blocking ones have been resolved by reviewer
-   [ ] Design OK
-   [x] Can be tested by POs
-   [ ] PR was introduced during daily meeting

### CODE REVIEW

File: libs/ui/package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "@ant-design/icons": ">=5.2",
     "@apollo/client": ">=3.8.1",
     "antd": "5.15.3",
-    "aristid-ds": "10.1.0-c112384",
+    "aristid-ds": "10.1.0-d627b25",
```
Comment (P0ppoff) [inline libs/ui/package.json:54]: **todo:** pas la même version que les autres

File: libs/ui/src/components/Explorer/edit-settings/SidePanel.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const {t} = useSharedTranslation();
+
+    // TODO: handle conflict with app side panel
+    const {activeSettings, onClose} = useEditSettings();
+
+    if (!activeSettings) {
+        return null;
+    }
+
+    // TODO: handle transition opening/closing on floating prop true
```
Comment (P0ppoff) [inline libs/ui/src/components/Explorer/edit-settings/SidePanel.tsx:19]: **todo:** Est ce que c'est le bon moment de le faire ?

File: libs/ui/src/components/Explorer/edit-settings/SidePanel.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// License text available at https://www.gnu.org/licenses/lgpl-3.0.txt
+import {ButtonHTMLAttributes, ComponentProps, FunctionComponent} from 'react';
+import {KitSidePanel} from 'aristid-ds';
+import {useEditSettings} from './useEditSettings';
+import {useSharedTranslation} from '_ui/hooks/useSharedTranslation';
+
+export const SidePanel: FunctionComponent = () => {
+    const {t} = useSharedTranslation();
+
+    // TODO: handle conflict with app side panel
```
[the remark this answers is not in the document]
Comment (P0ppoff) [reply libs/ui/src/components/Explorer/edit-settings/SidePanel.tsx:12]: later

File: libs/ui/src/components/Explorer/edit-settings/SidePanel.tsx (hunk 2 above)
Comment (P0ppoff) [reply libs/ui/src/components/Explorer/edit-settings/SidePanel.tsx:19]: emile en parle au UX

File: apps/data-studio/src/components/Router/Router.tsx [hunk 4]
```diff
@@ -45,11 +45,15 @@ const Router: FunctionComponent = () => {
...
-                        </Content>
+                        <Explorer.EditSettingsContextProvider>
+                            <Content style={{background: themeVars.defaultBg, overflow: 'hidden'}}>
+                                <UserPanel userPanelVisible={userPanelVisible} hideUserPanel={hideUserPanel} />
+                                <NotifsPanel setNbNotifs={_setNbNotifs} />
+                                <Routes />
+                            </Content>
+                            {/* TODO: handle conflict with app side panel */}
+                            <Explorer.SettingsSidePanel />
+                        </Explorer.EditSettingsContextProvider>
```
Comment (P0ppoff) [inline apps/data-studio/src/components/Router/Router.tsx:56]: **note:** Voici comment on intègre le composant`<Explorer />` dans **Data-studio**. Il manque juste l'appel du composant en tant que tel qui est plus bas dans l'arborescence.

**Review by philippechevieux — APPROVED** [review]

**Pull request merged**